File: 2026/04-April/15-Wednesday/Solution.c
#include <stdio.h>
#include <string.h>
#include <limits.h>
#include <stdlib.h>

int closestTarget(char** words, int wordsSize, char* target, int startIndex) {
    int ans = INT_MAX;

    for (int i = 0; i < wordsSize; i++) { 
        if (strcmp(words[i], target) == 0) {
            int d = abs(i - startIndex);
            int minDist = d < (wordsSize - d) ? d : (wordsSize - d);
            if (minDist < ans) ans = minDist;
        }
    }

    return ans == INT_MAX ? -1 : ans;
}

File: 2026/04-April/15-Wednesday/solution.py
class Solution:
    def closestTarget(self, words, target, startIndex):
        n = len(words)
        ans = float('inf')
        
        for i in range(n):
            if words[i] == target:
                d = abs(i - startIndex) 
                ans = min(ans, min(d, n - d))
        
        return -1 if ans == float('inf') else ans
